### ISSUE TITLE
MySQL backup uses invalid filename in tar

### ISSUE THREAD

**Issue by mswiege** [simatec/ioBroker.backitup#725]
**Describe the bug**  
The backup of my MySQL DB results in a mysql....tar.gz file. After unzipping, you get the mysql.tar file (as expected). But in this tar file is just one file (the SQL file) with the name ".". Such a filename messes with the Linux tar command. The tar file can't be extracted because a file with the name "." is invalid in Linux. On Windows, I can extract the file using 7zip, which automatically renames the file to "_" during extraction.

**To Reproduce**  
I don't see anything fancy in my backup settings. It uses '/usr/bin/mysqldump' and uploads the file to Google Drive. 
My DB is quite big, the extracted SQL file is ~650MB.

**Expected behavior**  
Use a name like iobroker.sql as the filename that will be in the tar file.

**Versions:**  
 - Adapter version: 2.4.10
 - JS-Controller version: 4.0.23
 - Node version: v14.20.0 
 - Operating system: linux (raspberry 4)

**Comment by simatec** [comment]
This designation was deliberately chosen and is also required by Backitup for a restore

**Comment by mswiege** [comment]
But can't you deliberately choose a filename which results in a valid tar archive? Maybe the restore function can check for both, the old filename "." and the new filename "iobroker.sql" and uses whichever it finds for the restore.

**Comment by Kettenknecht** [comment]
This is an absolute stupid behavior. For i don`t know what reasons the backup adapter doesnt restore my sql data. No i fiddling around with this broken archive with this idiotic filename. Nice backup!

**Comment by simatec** [comment]
I ask for reasonable and respectful communication

**Comment by Kettenknecht** [comment]
So respect common standards for your software, and don't bring your users into trouble. Even restore doesn't work, nor unpacking of the archive file. Using the character which means "the actual directory" for filenname in tar files is what i said.

**Comment by simatec** [comment]
a restore with backup is possible without problems

**Comment by Kettenknecht** [comment]
Close the issue. There is nothing to see :o))

**Comment by simatec** [comment]
I think you should rethink the way you communicate here.
You surely know that iobroker is an open source project and all developers do it as a hobby in their spare time.

None of the developers wants to be insulted with such improper comments.

I will close this issue when it is done. Until then it will remain open.

**Comment by Kettenknecht** [comment]
This is no apologize for stealing users their time, and perhaps also data, with software like yours. Why you don't change the issue? Maybe you don't understand it at all? Have you ever seen an Unix shell? What's the purpose of this behavior, to make a backup as tar file, with the filename "." in it? Did you understand what this causes if you want to check the backup, or use the data in another db outside of ioBroker? I don't think so. Your second answer in this thread suggests, you cargoculted "your" code without understanding anything. And your attitude "backups are working without problem" is next. No, they don't. And there's even no error message.

**Comment by mswiege** [comment]
@Kettenknecht, sounds to my like you are not understanding how open source software works.
How can you be so disrespectful to an open source Developer of the software you are using (for free!)? He is not forcing you to use his software.

"Why you don't change the issue?" - maybe he has no time? Maybe he just prefers laying in the sun? ... Why would he? Do you pay him in any way to do it? Do you have any contract with him that obligates him to do anything with this software?

"Maybe you don't understand it at all? Have you ever seen an Unix shell?" - looking at his code contributions and comparing it with yours (0) I assume he has much more knowledge about software development than you do. If you are such an expert, why not fix it yourself and provide a pull request?

You really should overthink your attitude and the way you talk to people.

@simatec Thank you for writing this adapter! I hope you find the time in the future to fix this issue because it is simply suboptimal. While your adapter seem to have no issue with this, it would be nice to make it possible to easily extract the backuped data without the adapter (on a Unix system).

**Comment by simatec** [comment]
@mswiege I will have a look at what could be changed there

**Comment by mswiege** [comment]
Just had a glimpse at the code and I don't understand why this happens. To me the code looks like the SQL file should actually be called `mysql_${getDate()}_backupiobroker.sql` and not `.` in the tar. :-)

**Comment by Kettenknecht** [comment]
I want to apologize. If you stand at night with such an unecessary problem, after other unecessary problems after an Ubuntu update, and read an answer, which means "all okay so", you can get a bit furios. But i don´t wanted to offend you. Sorry again.

**Comment by simatec** [comment]
@mswiege 
Can you please install the latest dev version via the following Github link?
https://github.com/simatec/ioBroker.backitup/tarball/dev


There please create a backup and see if the filename fits and a restore would also be good to test.
unfortunately i don't have mysql for testing at the moment.

**Comment by simatec** [comment]
![image](https://user-images.githubusercontent.com/39792461/182614845-2d594af6-0e0b-4266-982a-3e2d8b4255de.png)

**Comment by mswiege** [comment]
I'm currently not at home, that's also why I could only have a glimpse at the code yet.
I might be able to check it (quite late) in the evening today and will report how it went.

**Comment by simatec** [comment]
A test would be super.
Please test backup and restore

**Comment by mswiege** [comment]
Was able to do a first test on my mobile phone via VPN already. 😁

I made a backup and the tar file now contains a proper file (mysql_...backupiobroker.sql) but the file is in a folder structure:
`opt/iobroker/mysql_...backupiobroker.sql`
was this intended?

I couldn't test the restore yet.

**Comment by simatec** [comment]
It should not be like this ... Wait I fix that, then you can update again from the DEV Branch and test it

**Comment by simatec** [comment]
@mswiege The filename is fixed and can be tested in the DEV branch

**Comment by mswiege** [comment]
Still got the same thing.

**Comment by simatec** [comment]
Please make a Upload after install an restart the Adapter

**Comment by simatec** [comment]
@mswiege Have you been able to test it yet?

**Comment by mswiege** [comment]
Works now. The tar now has a file called mysql_...backupiobroker.sql which is easy to extract via terminal and also contains the valid mysqldump.

The only thing I didn't test is if the actual "restore" button in the frontend to rebuild the database works (because I'm a bit afraid to do this in my production system).

**Comment by simatec** [comment]
A restore would of course still be important to test before I can pack it into the next release.  

Unfortunately I don't have mysql to test at the moment.

**Comment by mswiege** [comment]
I will setup a test system in a VM to try the restore. Might take a few days until I'm done because I don't have much time at the moment.

**Comment by mswiege** [comment]
Hey @simatec , your code changes look good. 
I tried a restore of mysql from an old backup (tar containing file ".") and a restore from a new backup (with proper file name in tar). Both restored the DB without issues.

Some details of my test:
- Tested in a fresh iobroker installation (in a debian VM) with mysql running on same host
- I used the frontend in admin to restore the mysql backups (not via command line)
- Only tested the mysql restore (no iobroker,... or a different DB type)
